[inetstack] add udp integration tests

anandbonde opened the pull request:
1. Moved some layer 4 udp tests into integration tests.
2. Cleaned up unnecessary comments and renamed for clarity.
3. Renamed tcp-test directory to make it plural.
4. Fixed bug: setsockopt should not happen for UDP sockets.

iyzhang commented on nettest/input/linux/udp/bind-addr-in-use.pkt:
Can we move this test to the integration tests?

anandbonde replied:
Sure, do you mean, keep them in code instead of .pkt files?

anandbonde replied:
Done.

iyzhang commented on nettest/input/windows/udp/bind-addr-in-use.pkt:
Can we move this test to the integration tests?

anandbonde replied:
Sure, do you mean, keep them in code instead of .pkt files?

anandbonde replied:
Done.

iyzhang commented on src/rust/inetstack/protocols/layer4/udp/tests.rs:
```diff
@@ -451,87 +420,6 @@ fn udp_loop2_ping_pong() -> Result<()> {
     Ok(())
 }
 
-//======================================================================================================================
```
Let's move these three tests to the integration tests. We should create a tests/rust/udp-test directory and move these tests to a bind/mod.rs. Then we will not need a separate directory for windows and linux in nettest, which is not designed to handle error cases.

anandbonde replied:
Done.

anandbonde committed:
[catnip] Enhancement: moved some udp layer4 tests to integration tests

[catnap] Bug Fix: don't set sockopts for non-tcp

The set_linger sockopt was being set for a UDP socket too. This was
causing integration tests to fail.

iyzhang commented on linux.mk:
```diff
@@ -214,21 +196,17 @@ export MTU ?= 1500
 export PEER ?= server
 export TEST ?= udp-push-pop
-export TEST_INTEGRATION ?= tcp-test
+export TEST_INTEGRATION ?= tcp-tests
```
Should this also include udp-tests?

anandbonde replied:
This is the default value for the integration test. This line changed becuase I changed `tcp-test` to `tcp-tests` (plural). The CI pipeline does run `udp-tests`.
